APMrover2/cruise_learn.cpp

```diff
@@ -0,0 +1,51 @@
+#include "Rover.h"
+
+// start cruise throttle and speed learning
+void Rover::cruise_learn_start()
+{
+    // if disarmed do nothing
+    if (!arming.is_armed()) {
+        cruise_learn.learning = false;
+        gcs().send_text(MAV_SEVERITY_CRITICAL, "Cruise Learning NOT stated");
+        return;
+    }
+    // when switch is high, start learning
+    float speed;
+    if (g2.attitude_control.get_forward_speed(speed)) {
```

Comment:
miss a else to said that we don't have good speed estimation so learning isn't started